### PR TITLE
SELV3-596: Enable adding data element mapping

### PR DESCRIPTION
**Pull request by mdulko**

### CODE REVIEW

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        {name: 'Opening Balance', value: 'Opening Balance'},
+        {name: 'Received', value: 'Received'},
+        {name: 'Closing Balance', value: 'Closing Balance'},
+        {name: 'Total Negative Adjustments', value: 'Total Negative Adjustments'},
+        {name: 'Total Positive Adjustments', value: 'Total Positive Adjustments'}
+    ];
+
+    const [productOptions, setProductOptions] = useState([]);
+
+    const [providedName, setProvidedName] = useState("")
```
Comment (sniedzielski) [inline src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx:24]: please add semicolons at the end of lines: 24,25,26,27

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        () => {
+            return getService('adminDhis2');
+        },
+        []
+    );
+
+    const fetchDataOrderables = () => {
+        serverService.getDataOrderables()
+            .then((fetchedOrderables) => {
+                const { content } = fetchedOrderables
```
Comment (sniedzielski) [inline src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx:46]: add semicolon at the end of line

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                    name: element.fullProductName,
+                    value: element.fullProductName
+                }));
+                setProductOptions(elements);
+
+            });
+    }
+
+    useEffect(() => {
+        fetchDataOrderables()
```
Comment (sniedzielski) [inline src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx:58]: semicolon at the end of file

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+            name: providedName,
+            source: selectedIndicatorType,
+            indicator: selectedIndicator,
+            orderable: selectedProduct,
+            element: `${selectedIndicator} - ${selectedProduct}`
+        }
+
+        serverService.addDataElement(serverId, datasetId, element)
+            .then(() => {
+                refetch()
```
Comment (sniedzielski) [inline src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx:72]: semicolons on lines:
72,73, 75

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
+import React, { useMemo, useEffect, useState } from 'react';
+import { toast } from 'react-toastify';
+
+import getService from '../../../react-components/utils/angular-utils';
+import { SearchSelect } from '../../../requisition-order-create/search-select'
+
+function AdminDhis2DataElementForm({onSubmit, onCancel, refetch, serverId, datasetId}) {
```
Comment (sniedzielski) [inline src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx:7]: `function AdminDhis2DataElementForm({ onSubmit, onCancel, refetch, serverId, datasetId }) {`

Comment (sniedzielski) [reply src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx:7]: simply add the white space after and before curly bracket.

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+    const [productOptions, setProductOptions] = useState([]);
+
+    const [providedName, setProvidedName] = useState("")
+    const [selectedProduct, setSelectedProduct] = useState("")
+    const [selectedIndicator, setSelectedIndicator] = useState("")
+    const [selectedIndicatorType, setSelectedIndicatorType] = useState("")
+
+    const setInitialValues = () => {
+        setProvidedName('')
```
Comment (sniedzielski) [inline src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx:30]: I'm not sure, but semicolons are also necessary here on lines: 30,31,32,33

**Review by sniedzielski — COMMENTED** [review]
@mdulko I left some minor comments, please refer to them.

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx (hunk 1 above)
Comment (mdulko) [reply src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx]: updated

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx (hunk 2 above)
Comment (mdulko) [reply src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx]: updated

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx (hunk 3 above)
Comment (mdulko) [reply src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx]: updated

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx (hunk 4 above)
Comment (mdulko) [reply src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx]: updated

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx (hunk 5 above)
Comment (mdulko) [reply src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx]: updated

File: src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx (hunk 6 above)
Comment (mdulko) [reply src/admin-dhis2/components/AdminDhis2DataElementForm/AdminDhis2DataElementForm.jsx]: updated

**Review by sniedzielski — APPROVED** [review]
LGTM

**Pull request merged**